src/web/server/document.tsx

```diff
@@ -61,6 +61,7 @@ export const document = ({ data }: Props) => {
      */
     const priorityScripts = [
         polyfillIO,
+        getDist({ path: 'commercial.js', legacy: false }),
```

Comment:
this needs to run before the commercial frontend>dcr bundle. 

but that bundle doesn't have module/no-module builds, so it can't be moved to the `priorityLegacyScripts`/`priorityNonLegacyScripts` arrays, otherwise modern browsers try to load it a module and break.

hence loading it like this. which feels icky – any suggestions?

Reply:
Thinking building the scripts needs a slight refactor along the lines of:

```js
const priorityScriptTags = [{
  'path': 'one',
  'legacy': true
},
{
  'path': 'two'
},
{
  'path': 'three',
  'legacy': false
}];



const mapToScriptTags = (pathObj) => {
  if (pathObj.legacy === undefined) {
    return `Normal script tag: ${pathObj.path}`
  } else if (pathObj.legacy === true) {
    return `Script tag with momodule: ${pathObj.path}`
  } else {
    return `Script tag with type="module": ${pathObj.path}`
  }
}

// Insert these into the page instead
console.log(priorityScriptTags.map(mapToScriptTags).join('\n'))
```

So the ordering is maintained through one array but we maintain the module / nomodule and normal script tag mechanism:

```js
`Script tag with momodule: one
Normal script tag: two
Script tag with type="module": three`
```

Does that make sense?

Reply:
Maybe even more simplified:

```js
const priorityScriptTags = [{
  'path': 'one',
  'hasLegacy': true
},
{
  'path': 'two',
  'hasLegacy': false
},
{
  'path': 'three',
  'hasLegacy': true
}];



const mapToScriptTags = (pathObj) => {
  if (pathObj.hasLegacy === false) {
    return `Normal script tag: ${pathObj.path}`
  } else if (pathObj.hasLegacy === true) {
    return `
Script tag with momodule: ${pathObj.path}
Script tag with type="module": ${pathObj.path}
`
  }
}

console.log(priorityScriptTags.map(mapToScriptTags).join('\n'))
```

Reply:
what is wrong with calling it as a `module/no-module`? if we compile the code here and it gets build in our webpack config, then it should be fine?

Reply:
we use `module/no-module` purely as a way of checking if the browser supports those features, its an indication that the browser is modern enough to support a build that used `@babel/preset-modules` rather than `@babel/preset-env`

Reply:
Hey @liywjl I think @sndrs is referring to the Commercial JS bundle that we get passed a url to from Frontend (via CAPI json). So it's not a bundled file.